src/DataShare/DataShare/DataSet/RemoveAzDataShareDataSet.cs

```diff
@@ -0,0 +1,152 @@
+
+        /// <summary>
+        /// Data share name.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Azure data share name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string ShareName { get; set; }
+
+        /// <summary>
+        /// Data set name.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data set name.",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.DataSet, "ResourceGroupName", "AccountName", "ShareName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The resourceId of the azure data set.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of the azure data set.",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ResourceIdCompleter(ResourceTypes.DataSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Azure data set object.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.ObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "The azure data set object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDataShareDataSet InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Return object (if specified).")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string resourceId = null;
+
+            if (this.ParameterSetName.Equals(
+                ParameterSetNames.ResourceIdParameterSet,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                resourceId = this.ResourceId;
+            }
+
+            if (this.ParameterSetName.Equals(ParameterSetNames.ObjectParameterSet, StringComparison.OrdinalIgnoreCase))
+            {
+                resourceId = this.InputObject.Id;
+            }
+
+            if (!string.IsNullOrEmpty(resourceId))
+            {
+                var parseResourceId = new ResourceIdentifier(resourceId);
+                this.ResourceGroupName = parseResourceId.ResourceGroupName;
+                this.AccountName = parseResourceId.GetAccountName();
+                this.ShareName = parseResourceId.GetShareName();
+                this.Name = parseResourceId.GetDataSetName();
+            }
+
+            this.ConfirmAction(
```

Comment:
@gaurangisaxena same comment -- please use `ShouldProcess` instead of `ConfirmAction`

This comment applies to all instances of this in the PR.

Reply:
Same comment, this is fine.